src/zope/app/testing/dochttp.py

```diff
@@ -59,17 +60,17 @@
 
     ]
 
-def dochttp(args=sys.argv[1:], default=None):
+def dochttp(args=sys.argv[1:], default=None, output_fp=None):
     """Convert a tcpwatch recorded sesssion to a doctest file"""
     if default is None:
         default = default_options
 
-    options, args = parser.parse_args(default+args)
+    options, args = parser.parse_args(default + args)
     try:
         directory, = args
     except:
         parser.print_help()
-        sys.exit(1)
+        raise
```

Comment:
No?  The ValueError: need more/too many values to unpack is not particularly illuminating, and the traceback is not helpful.  TBH I'd rather do `parser.error('Exactly one argument expected: directory')` instead of print_help+sys.exit.

All this is doing is making sure there's one and only one positional argument on the command line, since optparse can't do that for us.  (If we switched to argparse, we wouldn't need this try/except.)